homeassistant/components/tuya/base.py

```diff
@@ -31,25 +31,25 @@ class IntegerTypeData:
     @property
     def max_scaled(self) -> float:
         """Return the max scaled."""
-        return self.scale_value(self.max)
+        return self.max * self.step / (10**self.scale)
```

Comment:
Please reference officual resources that explain how step is involved here.
Documentation and/or reference implementation from Tuya (or other third-party apps).

As far as I am aware, this is not correct, not documented as such and haven't been able to find a single implementation that does this.

This is only found in certain (manufacturer-specific) implementations, like the moes thermostats.

Reply:
Here is a conversation an example, that is also incorrect:
<https://github.com/home-assistant/core/issues/66480#issuecomment-1040333643>

The formula proposed in this PR, will also not the diagnostics provided in that issue linked above. Different manufacturers, implemented something their own, just like Moes did.

Reply:
@frenck thanks for the hint. We could  adapt the code so that we use special implementations may be model specific and the default if there's no model specific... Something like 'model specific overrides'

Reply:
We don't allow tha. Device-specific code in our code base: Tuya has 400K+ SKUs, that would render our codebase unmaintainable.